src/hooks/useEditLabel.js

```diff
@@ -0,0 +1,17 @@
+import { useIntl } from 'react-intl';
+import { get } from 'lodash-es';
+
+export default function useEditLabel(fieldSchema, noAsterisk = false) {
+  const intl = useIntl();
+  let label = null;
+  if (fieldSchema?.editLabelId) {
+    label = intl.formatMessage({ id: fieldSchema.editLabelId });
+  } else if(fieldSchema?.labelId){
+    label = intl.formatMessage({ id: fieldSchema.labelId });
+  }else 
```

Comment:
dont forget to check for `editLabel`

Reply:
so it's a little weird to implement this in an abstract way (editLabel/viewLabel) but only to add functionality to the edit component TextEditor. the proper implementation of this should support all Edit and View components, otherwise it's just confusing

Reply:
like as a developer i would see the "viewLabelId" property is supported, then anywhere I tried to use it wouldn't work. does that make sense?